packages/@aws-cdk/aws-stepfunctions-tasks/test/codebuild/start-build.test.ts

```diff
@@ -56,7 +57,53 @@ test('Task with only the required parameters', () => {
   });
 });
 
-test('Task with all the parameters', () => {
+test('Task with env variables parameters', () => {
+  // WHEN
+  const task = new CodeBuildStartBuild(stack, 'Task', {
+    project: codebuildProject,
+    integrationPattern: sfn.IntegrationPattern.RUN_JOB,
+    overrides: {
+      environmentVariables: {
+        env: {
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+          value: 'prod',
+        },
+      },
+    },
+  });
+
+  // THEN
+  expect(stack.resolve(task.toStateJson())).toEqual({
```

Comment:
We have helpers like `expect(stack).toHaveResourceLike()`, I think you might want to use them here (take a look at the other tests in this module for inspiration).

Reply:
@skinny85 the unit test is checking the definition of state machine described by JSON-based Amazon States Language if it’s expected. Looks like `expect(stack).toHaveResourceLike()` only works for the CloudFormation resources.

Reply:
I mean, that JSON has to be in the CloudFormation resource too, right? It won't magically get to the StepFunctions API 🙂.

But that's fine, I don't care that much, I'll let @shivlaks decide what he wants here.

Reply:
you're right though @skinny85 - it is in the CloudFormation resource at some point, but that's a JSON escaped string of the entire state machine definition.

in the context of tasks, this is where we ended up settling on as it's a little more crisp.  I'm fine with it as it is 🙂